crates/turborepo-lib/src/rewrite_json.rs

```diff
@@ -0,0 +1,420 @@
+ * find the last node that exists with in this JSON key path.
+ */
+fn get_closest_node<'a>(
+    current_node: &'a jsonc_parser::ast::Value<'a>,
+    target_path: &[&'a str],
+    current_path: &'a mut Vec<&'a str>,
+) -> (&'a mut Vec<&'a str>, &'a jsonc_parser::ast::Value<'a>) {
+    // No target_path? We've arrived.
+    if target_path.is_empty() {
+        return (current_path, current_node);
+    }
+
+    match current_node {
+        // Only objects can have key paths.
+        jsonc_parser::ast::Value::Object(obj) => {
+            // Grab the last property (member) which matches the current target_path
+            // element.
+            let object_property = obj.properties.iter().rev().find(|property| {
+                let current_property_name = property.name.as_str();
+                target_path[0] == current_property_name
+            });
+
+            // See if we found a matching key. If so, recurse.
+            match object_property {
+                Some(property) => {
+                    let next_current_node = &property.value;
+                    let next_property_name = property.name.as_str();
+                    let next_current_path = &mut *current_path;
+                    next_current_path.push(next_property_name);
+                    let next_target_path = &target_path[1..];
+
+                    // Tail call!
+                    get_closest_node(next_current_node, next_target_path, next_current_path)
+                }
+                None => (current_path, current_node),
+            }
+        }
+        // All other node types are complete.
+        _ => (current_path, current_node),
+    }
+}
+
+/**
+ * Given path segments, generate a JSON object.
+ */
+fn generate_object(path_segments: &[&str], value: &str) -> String {
+    let mut output = String::new();
+    let length = path_segments.len();
+
+    for path in path_segments {
+        output.push('{');
+        output.push('\"');
+        output.push_str(path);
+        output.push('\"');
+        output.push(':');
+    }
+    output.push_str(value);
+    for _ in 0..length {
+        output.push('}');
+    }
+
+    output
+}
+
+/**
+ * Given path segments, generate a JSON object member with an optional
+ * trailing separator.
+ */
+fn generate_member(path_segments: &[&str], value: &str, separator: &str) -> String {
+    let (key, remainder) = path_segments.split_first().unwrap();
+    let object = generate_object(remainder, value);
+    format!("\"{key}\":{object}{separator}")
+}
+
+/**
+ * Given a JSONC document and an object traversal path, `unset_path` will
+ * return a minimally-mutated JSONC document with all occurrences of the
+ * specified path removed.
+ */
+pub fn unset_path(
+    json_document_string: &str,
+    path: &[&str],
+) -> Result<(bool, String), RewriteError> {
+    let root = get_root(json_document_string)?;
+
+    // The key path can appear multiple times. This a vec that contains each time it
+    // occurs.
```

Comment:
Could you add a test for this case? I'm having trouble understanding when we'll have multiple key paths.

Reply:
There's already a test for it (every single test case includes duplicated keys); but this is basically for the the situation where:

```json
{
  "thing": "hello, world",
  "thing": "also, hello."
}
```

If you try to unset `thing` and only get one of those you didn't actually succeed at unsetting it when deserialized. Deserialize, mutate (unset), serialize ensures that you get only one, but if you don't do deserialization you have to account for garbage-in.